### PR TITLE
RA-1634 : Update MetaData Packages in the MetaData Module

### PR DESCRIPTION
**Pull request by mozzy11**
Ticket : https://issues.openmrs.org/browse/RA-1634
Update Metadata Sharing packages in the Reference Metadata module

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/referencemetadata/ReferenceMetadataActivator.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ private void installConcepts() {
         	Context.flushSession(); //Flush so that purges are not deferred until after data import
         	
         	DataImporter dataImporter = Context.getRegisteredComponent("dataImporter", DataImporter.class);
-            dataImporter.importData("Reference_Application_Concepts-24.xml");
+            dataImporter.importData("Reference_Application_Concepts-26.xml");
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/referencemetadata/ReferenceMetadataActivator.java]: Any reason why this is 26 instead of 25?

Comment (mozzy11) [reply api/src/main/java/org/openmrs/module/referencemetadata/ReferenceMetadataActivator.java]: sorry , it was a mistake

Comment (mozzy11) [reply api/src/main/java/org/openmrs/module/referencemetadata/ReferenceMetadataActivator.java]: corrected it

**Comment by dkayiwa** [comment]
Are these the only packages that changed?

**Comment by mozzy11** [comment]
yes , they are the only that am required to update as per this 
https://wiki.openmrs.org/display/docs/OpenMRS+Reference+Application+Release+Process

**Comment by mozzy11** [comment]
@dkayiwa  , waiting for your comments

**Comment by dkayiwa** [comment]
Which openmrs platform version did you use to generate these changes on the mdsbuilder server?

**Comment by mozzy11** [comment]
@dkayiwa   mdsBuilder runs on platform 2.0.6 , so that what used

**Comment by dkayiwa** [comment]
And what is the name of the CIEL file that you used?

**Comment by mozzy11** [comment]
openmrs_concepts_2.0_20190701.sql

**Comment by dkayiwa** [comment]
And which OpenMRS platform version will the reference application that you are releasing be running on?

**Comment by mozzy11** [comment]
platform 2.2. Does that mean we have to upgrade the platform version for the mdsBuilderServer ?

**Comment by mozzy11** [comment]
ok . can you do that for me then? i dont have permisons to do that

**Comment by dkayiwa** [comment]
Do you have ssh access to that server?

**Comment by mozzy11** [comment]
yes i do .

**Comment by dkayiwa** [comment]
So what kind of permissions are you looking for?

**Comment by mozzy11** [comment]
oh i ddnt know i could also upgrade the platform with ssh access. On which server is it hosted , i mean if i can do it via UI . ,it may be more hectic for me to use commands to upgrade the platform.

**Comment by mozzy11** [comment]
does it have the same directories as our local instances we run with tomcat ??

**Comment by mozzy11** [comment]
the server seems to be already running on pltform 2.2

here i the distro properties file
```
 version=1.0.0
distro.platform=2.2.0
omod.metadatasharing=1.3.0
omod.metadatamapping=1.3.3
omod.dataexchange=1.3.3
omod.legacyui=1.3.3

```

**Comment by mozzy11** [comment]
@dkayiwa  i updated the Metadata Packages ,using  **openmrs_concepts_2.2_20191021.sql**  for platform 2.2.
I updated some module versions in the POM , because i was getting alot datase of errors , so i solved thatby updating the Core version dependence to 2.2 and some other module versions

**Comment by dkayiwa** [comment]
Did you see the travis build failure?

**Comment by mozzy11** [comment]
@dkayiwa  , resolved the error

**Comment by mozzy11** [comment]
@dkayiwa  , any comments here ?

**Comment by mozzy11** [comment]
ping @dkayiwa

**Comment by mozzy22** [comment]
@dkayiwa  , some look into this

File: api/src/test/resources/BuiltInReportsTestDataset.xml [hunk 2]
```diff
@@ -114,7 +114,7 @@
 						date_created="2016-07-07 08:14:50" voided="0" uuid="d16f8f93-83a3-4027-af42-3e2414f4b48b"/>
 
 	<obs obs_id="1" person_id="52" concept_id="1" encounter_id="1" obs_datetime="2015-06-18 22:21:32" location_id="1" value_coded="1" creator="1"
-	     date_created="2017-06-18 23:45:34" voided="0" uuid="8d3a3ea9-c8fd-4b7b-b6a8-6410500ad410"
+	     date_created="2017-06-18 23:45:34" voided="0" uuid="8d3a3ea9-c8fd-4b7b-b6a8-6410500ad410" status="FINAL"
```
Comment (dkayiwa) [inline api/src/test/resources/BuiltInReportsTestDataset.xml]: Was the status necessary?

Comment (mozzy22) [reply api/src/test/resources/BuiltInReportsTestDataset.xml]: yes , since i updated most of the modules , it is now required, it cant be null

Comment (dkayiwa) [reply api/src/test/resources/BuiltInReportsTestDataset.xml]: What error do you get when you do not touch it?

Comment (mozzy22) [reply api/src/test/resources/BuiltInReportsTestDataset.xml]: some thing like "column OBS.STATUS can not be null."

Comment (mozzy22) [reply api/src/test/resources/BuiltInReportsTestDataset.xml]: i have also confirmed in an actual running database , "column OBS.STATUS" cant be NULL

**Comment by dkayiwa** [comment]
After running these changes, do you have the `Head circumference` concept in the dictionary?

**Comment by mozzy22** [comment]
oh well i havent run them locally yet, let me run them and i see

**Comment by mozzy22** [comment]
@dkayiwa  , its not there. However it exists in the Dictionary on the mdsBuilder. Is there a criteria that the  Metadata Sharing module follows to select out concepts to be included in the  package for Reference Application Concepts ??

**Comment by dkayiwa** [comment]
Before you export new package versions on mdsbuilder, you need to select and include.

**Comment by mozzy22** [comment]
@dkayiwa ,  i have included the Head Circumfrenece Cocept

**Comment by dkayiwa** [comment]
@mozzy11 i mean adding it during the new package version process of the user interface.

**Comment by mozzy22** [comment]
yes @dkayiwa  , thats where i included it in when i was creating the new package, before i exported the package

**Comment by dkayiwa** [comment]
Have you now compiled, run the module, and confirmed it now exists?

**Comment by mozzy22** [comment]
in other words , i deleted the previous package i had created from the mdsBuider server, and created a new one which includes the Head circumference concept

**Comment by dkayiwa** [comment]
Is that the only new concept that the new refapp requires?

**Comment by mozzy22** [comment]
well , i would think so. Am not aware of any new feature that would require any new concept as far as the Features Road Map we have worked on is concerned

**Comment by dkayiwa** [comment]
Have you now compiled, run the module, and confirmed it now exists?

**Comment by mozzy22** [comment]
its what am doing now ..

**Comment by mozzy22** [comment]
yes @dkayiwa , the Head Circumfrence Concept exist , ive just ran the latest snaphot version of the reff app , and deployed the Module ive compliled

**Comment by dkayiwa** [comment]
Is that a fresh installation or an existing one?

**Comment by mozzy22** [comment]
a fresh installation

**Comment by dkayiwa** [comment]
Can you also test an existing installation?

**Comment by mozzy22** [comment]
like i deloy the module in an existing installation without first deleting the database ?

**Comment by dkayiwa** [comment]
Any instance of the reference application that was running without your changes.

**Comment by mozzy22** [comment]
it doesnt update concepts in an existing installation

**Comment by dkayiwa** [comment]
So can you fix that?

**Comment by mozzy11** [comment]
@dkayiwa  i fixed that , i increased the  **METADATA_VERSION**  to 17;

**Comment by mozzy11** [comment]
it now works on an existing installtion

**Comment by dkayiwa** [comment]
What is the difference between this pull request and the previous one? https://github.com/openmrs/openmrs-module-referencemetadata/pull/46

**Comment by mozzy11** [comment]
@dkayiwa  , with this PR ,
*  i built and updated the packages in this PR using mdsBuilder server , not manual editing like the other side. That saves alot of human error especially when it comes to mappings and UUIDs , no wonder we got CI errors with the other Pr.
I shud actually close it.

* I built this packages with the latest Updated Dictionary (not sure where some updates are also made in the already existing Concepts) , but its the latest Dictionary anyway

**Pull request merged**